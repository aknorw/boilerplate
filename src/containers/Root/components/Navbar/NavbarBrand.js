import styled from 'styled-components'
import { Link } from 'react-router-dom'

export default styled(Link)`
  color: white;
  text-decoration: none;
  cursor: pointer;
  -webkit-font-smoothing: antialiased;
  font-size: 1.25em;
  font-weight: bold;
  text-transform: uppercase;
`
