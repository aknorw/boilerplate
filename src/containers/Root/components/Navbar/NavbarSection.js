import styled from 'styled-components'

export default styled.div`
  display: inline-flex;
  align-items: baseline;
  font-size: 1em;
  > *:not(:last-child) {
    margin-right: .5rem;
  }
`
