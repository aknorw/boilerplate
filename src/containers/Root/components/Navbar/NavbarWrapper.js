import styled from 'styled-components'

export default styled.nav`
  flex-shrink: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background-color: black;
  color: white;
`
