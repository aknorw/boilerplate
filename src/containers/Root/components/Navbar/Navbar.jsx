import React from 'react'
import PropTypes from 'prop-types'

import NavbarWrapper from './NavbarWrapper'
import NavbarBrand from './NavbarBrand'
import NavbarSection from './NavbarSection'
import NavLink from './NavLink'

const Navbar = ({ title }) => (
  <NavbarWrapper>
    <NavbarBrand to="/">
      {title}
    </NavbarBrand>
    <NavbarSection>
      <NavLink to="/settings">
        Settings
      </NavLink>
    </NavbarSection>
  </NavbarWrapper>
)

Navbar.propTypes = {
  title: PropTypes.string.isRequired,
}

export default Navbar
