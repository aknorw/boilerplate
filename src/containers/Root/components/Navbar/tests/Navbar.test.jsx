import React from 'react'
import { shallow } from 'enzyme'

import Navbar from '../Navbar'

console.error = jest.fn()

describe('<Navbar />', () => {
  it('should be defined', () => {
    expect(Navbar).toBeDefined()
  })
  it('should log 1 error when no props are passed', () => {
    shallow(<Navbar />)
    expect(console.error).toHaveBeenCalledTimes(1)
  })
  it('should render correctly', () => {
    const tree = shallow(<Navbar title="Test" />)
    expect(tree).toMatchSnapshot()
  })
})
