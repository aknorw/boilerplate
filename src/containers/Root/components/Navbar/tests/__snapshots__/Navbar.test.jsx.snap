// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render correctly 1`] = `
<styled.nav>
  <Styled(Link)
    to="/"
  >
    Test
  </Styled(Link)>
  <styled.div>
    <Styled(Link)
      to="/settings"
    >
      Settings
    </Styled(Link)>
  </styled.div>
</styled.nav>
`;
